x-pack/plugins/fleet/server/services/epm/elasticsearch/template/default_settings.ts

```diff
@@ -6,61 +6,14 @@
  */
 
 import { appContextService } from '../../../app_context';
-import type { Field, Fields } from '../../fields/field';
-
-const QUERY_DEFAULT_FIELD_TYPES = ['keyword', 'text', 'match_only_text', 'wildcard'];
-const QUERY_DEFAULT_FIELD_LIMIT = 1024;
-
-const flattenAndExtractFields = (fields: Fields, path: string = ''): Field[] => {
-  let newFields: Array<Pick<Field, 'name' | 'type' | 'default_field'>> = [];
-  fields.forEach((field) => {
-    const fieldName = path ? `${path}.${field.name}` : field.name;
-    newFields.push({
-      ...field,
-      name: fieldName,
-    });
-    if (field.fields && field.fields.length) {
-      newFields = newFields.concat(flattenAndExtractFields(field.fields, fieldName));
-    }
-  });
-  return newFields;
-};
 
 export function buildDefaultSettings({
```

Comment:
nit: Could rename this to `buildIlmSettings` or something as it's its only responsibility now

Reply:
I prefer to keep calling it default settings so that the unit test to check for default_field make sense.